lib/core/refresh.js

```diff
@@ -0,0 +1,295 @@
+  EXPIRED: 'EXPIRED',
+  REFRESH_EXPIRED: 'REFRESH_EXPIRED'
+})
+
+export class TokenExpirationStatus {
+  constructor (scheme) {
+    this.scheme = scheme
+    this.$auth = scheme.$auth
+  }
+
+  _calculateTokenStatus () {
+    const now = Date.now()
+    let tokenExpiresAt, refreshTokenExpiresAt
+
+    try {
+      tokenExpiresAt = this.getTokenExpiration()
+      refreshTokenExpiresAt = this.getRefreshTokenExpiration()
+
+      if (!tokenExpiresAt) {
+        return TokenExpirationStatusEnum.UNKNOWN
+      }
+    } catch (error) {
+      return TokenExpirationStatusEnum.UNKNOWN
+    }
+
+    // Give us some slack to help the token from expiring between validation and usage
+    const timeSlackMillis = 500
+    tokenExpiresAt -= timeSlackMillis
+    refreshTokenExpiresAt -= timeSlackMillis
+
+    // Token is still valid
+    if (now < tokenExpiresAt) {
+      return TokenExpirationStatusEnum.VALID
+    }
+
+    if (now > refreshTokenExpiresAt) {
+      return TokenExpirationStatusEnum.REFRESH_EXPIRED
+    }
+
+    return TokenExpirationStatusEnum.EXPIRED
+  }
+
+  // ---------------------------------------------------------------
+  // Token Expiration helpers
+  // ---------------------------------------------------------------
+
+  getTokenExpiration () {
+    const _key = this.scheme.options.tokenExpirationPrefix + this.scheme.name
+
+    return this.$auth.$storage.getUniversal(_key)
+  }
+
+  setTokenExpiration (token) {
+    const _key = this.scheme.options.tokenExpirationPrefix + this.scheme.name
+
+    return this.$auth.$storage.setUniversal(_key, token)
+  }
+
+  syncTokenExpiration () {
+    const _key = this.scheme.options.tokenExpirationPrefix + this.scheme.name
+
+    return this.$auth.$storage.syncUniversal(_key)
+  }
+
+  // ---------------------------------------------------------------
+  // Refresh Token Expiration helpers
+  // ---------------------------------------------------------------
+
+  getRefreshTokenExpiration () {
+    const _key = this.scheme.options.refreshTokenExpirationPrefix + this.scheme.name
+
+    return this.$auth.$storage.getUniversal(_key)
+  }
+
+  setRefreshTokenExpiration (token) {
+    const _key = this.scheme.options.refreshTokenExpirationPrefix + this.scheme.name
+
+    return this.$auth.$storage.setUniversal(_key, token)
+  }
+
+  syncRefreshTokenExpiration () {
+    const _key = this.scheme.options.refreshTokenExpirationPrefix + this.scheme.name
+
+    return this.$auth.$storage.syncUniversal(_key)
+  }
+
+  updateTokenExpiration (result = {}) {
+    const token = this.scheme.$auth.getToken(this.scheme.name)
+    let decodedToken, tokenExpiration
+    const _tokenIssuedAtMillis = getProp(result, this.scheme.options.issuedAt) || Date.now()
+    const _tokenTTLMillis = (getProp(result, this.scheme.options.expiresIn) || this.scheme.options.token.maxAge) * 1000
+    const _tokenExpiresAtMillis = getProp(result, this.scheme.options.expiresAt) || _tokenIssuedAtMillis + _tokenTTLMillis
```

Comment:
Is `expiresAt` seconds or millis?

Reply:
It should be milliseconds. My idea is that `getProp(result, this.scheme.options.expiresAt)` would return the expiration date. Dates are in milliseconds, right?

Reply:
I simplified this method in commit https://github.com/nuxt-community/auth-module/pull/548/commits/5e57bd4bacf46d4ed92d8183391d39ddfdd3d8e9